### PR TITLE
Refactor CosmosTransferProcessStore extension to use newly introduce CosmosDbApi and add unit tests

### PR DESCRIPTION
**Pull request by bscholtes1A**

### CODE REVIEW

File: extensions/azure/transfer-process-store-cosmos/src/main/java/org/eclipse/dataspaceconnector/transfer/store/cosmos/CosmosTransferProcessStore.java [hunk 1]
```diff
@@ -152,48 +125,28 @@ public TransferProcess find(String id) {
...
         CosmosItemRequestOptions options = new CosmosItemRequestOptions();
         //todo: configure indexing
         var document = TransferProcessDocument.from(process, partitionKey);
-        try {
-            var response = failsafeExecutor.get(() -> container.createItem(document, new PartitionKey(partitionKey), options));
-            handleResponse(response);
-        } catch (CosmosException cme) {
-            throw new EdcException(cme);
-        }
+        failsafeExecutor.run(() -> cosmosDbApi.createItems(Collections.singletonList(document)));
```
Comment (paullatzelsperger) [inline extensions/azure/transfer-process-store-cosmos/src/main/java/org/eclipse/dataspaceconnector/transfer/store/cosmos/CosmosTransferProcessStore.java:134]: should be `saveItem`

File: extensions/azure/cosmos-common/src/main/java/org/eclipse/dataspaceconnector/cosmos/azure/CosmosDbApi.java [hunk 2]
```diff
@@ -23,9 +27,7 @@
 
     Stream<Object> queryItems(String query);
 
-    void deleteItem(String id);
-
     void createItems(Collection<CosmosDocument<?>> definitions);
```
Comment (paullatzelsperger) [inline extensions/azure/cosmos-common/src/main/java/org/eclipse/dataspaceconnector/cosmos/azure/CosmosDbApi.java:30]: can we rename this ` saveItems`?

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]
The only major thing is the `@JsonUnwrapped` annotation on the `CosmosDocument`. 
We either need to make sure all the queries in all other CosmosDB-based stores still work, or we put a pin in it, revert it and solve it in a subsequent PR.

**Comment by paullatzelsperger** [comment]
This patch should take care of the @JsonUnwrapped stuff: https://gist.github.com/paullatzelsperger/6b467769958adc4c860bd76b1b8ca46f

File: extensions/azure/transfer-process-store-cosmos/src/main/java/org/eclipse/dataspaceconnector/transfer/store/cosmos/CosmosTransferProcessStore.java (hunk 1 above)
Comment (bscholtes1A) [reply extensions/azure/transfer-process-store-cosmos/src/main/java/org/eclipse/dataspaceconnector/transfer/store/cosmos/CosmosTransferProcessStore.java]: done

File: extensions/azure/contract-definition-store-cosmos/src/main/java/org/eclipse/dataspaceconnector/contract/definition/store/model/ContractDefinitionDocument.java [hunk 3]
```diff
@@ -14,21 +14,24 @@
...
 import org.eclipse.dataspaceconnector.cosmos.azure.CosmosDocument;
 import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
 
 @JsonTypeName("dataspaceconnector:contractddefinitiondocument")
 public class ContractDefinitionDocument extends CosmosDocument<ContractDefinition> {
 
-    @JsonCreator
-    public ContractDefinitionDocument(@JsonProperty("wrappedInstance") ContractDefinition contractDefinition) {
-        //todo: lets think about whether this a good partition key
+    protected ContractDefinitionDocument() {
```
[the remark this answers is not in the document]
Comment (bscholtes1A) [reply extensions/azure/contract-definition-store-cosmos/src/main/java/org/eclipse/dataspaceconnector/contract/definition/store/model/ContractDefinitionDocument.java]: outdated

File: extensions/azure/cosmos-common/src/main/java/org/eclipse/dataspaceconnector/cosmos/azure/CosmosDbApi.java [hunk 4]
```diff
@@ -11,6 +11,10 @@
 
     void saveItem(CosmosDocument<?> item);
 
+    void updateItem(Object item);
```
[the remark this answers is not in the document]
Comment (bscholtes1A) [reply extensions/azure/cosmos-common/src/main/java/org/eclipse/dataspaceconnector/cosmos/azure/CosmosDbApi.java]: corrected

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**